### PR TITLE
Statsgenerator biotype

### PR DESCRIPTION
**Pull request by tgrego**
## Requirements

- Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion;
- Review the [contributing guidelines](https://github.com/Ensembl/ensembl/blob/release/90/CONTRIBUTING.md#why-could-my-pull-request-be-rejected) for this repository; remember in particular:
    - do not modify code without testing for regression
    - provide simple unit tests to test the changes
    - if you change the schema you must patch the test databases as well, see [Updating the schema](https://github.com/Ensembl/ensembl/blob/release/90/CONTRIBUTING.md#updating-the-schema)
    - the PR must not fail unit testing

## Description

_One or more sentences describing in detail the proposed changes._

Changes required to close ENSCORESW-2633
Added new method to BiotypeAdaptor, to be used in production pipelines.
fetch_all_by_group_object_db_type() allows fetching of the biotypes of a given biotype group.

## Use case

_Describe the problem. Please provide an example representing the motivation behind the need for having these changes in place._

To replace direct queries in production pipelines to the production schema with core API calls.

## Benefits

_If applicable, describe the advantages the changes will have._

Might be useful elsewhere.

## Possible Drawbacks

_If applicable, describe any possible undesirable consequence of the changes._

## Testing

_Have you added/modified unit tests to test the changes?_

Yes.

_If so, do the tests pass/fail?_

Pass, of course.

_Have you run the entire test suite and no regression was detected?_

Yes.

### CODE REVIEW

File: modules/Bio/EnsEMBL/DBSQL/BiotypeAdaptor.pm [hunk 1]
```diff
@@ -194,4 +194,39 @@ sub fetch_all_by_object_type {
...
+  Exceptions : none
+
+=cut
+
+sub fetch_all_by_group_object_db_type {
+  my ($self, $biotype_group, $object_type, $db_type) = @_;
+
+  $db_type //= 'core';
+
+  my $constraint = "b.biotype_group = ? AND b.object_type = ? AND FIND_IN_SET( ? , db_type)>0 ";
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/DBSQL/BiotypeAdaptor.pm]: This one isn't going to work with SQLite.

Comment (tgrego) [reply modules/Bio/EnsEMBL/DBSQL/BiotypeAdaptor.pm]: usually test with mysql so missed that...
Have updated the query and should now work in both, though not as elegant.

**Comment by nerdstrike** [comment]
Looks good! Shame about the set operation, but the alternative to a generic query is a lot more code.

**Comment by lairdm** [comment]
Seems reasonable, glad Kieron spotted the sqllite issue.

**Pull request merged**